### ISSUE TITLE
Adding The Parasite Tracker TCN Taxonomy to Global Names

### ISSUE THREAD

**Issue by campmlc** [GlobalNamesArchitecture/dwca_hunter#31]
@dimus  
The Arctos community would like to explore the possibility of adding ectoparasite taxonomy in use by institutions participating in the Parasite Tracker TCN to Global Names. The TPT is working to assemble taxonomy reference files for major groups of ectoparasites, with the names and classifications in csv format. Please let me know what steps would be involved with integrating these with Global Names.

**Comment by dimus** [comment]
@campmlc  can you tell a little more about the data source? Most importantly, where can I download the datataset. Do the curators of The Parasite Tracker consider their data open for other projects?

I am rewriting name resolution/reconciliation service at the moment so in normal curcumstances I would harvest this resource somewhere in November. Please let me know if there are reasons for me to put the harvest of these data as a higher priority.

**Comment by dimus** [comment]
@debpaul, the purpose is to create a link between Arctos and Parasite Tracker data. @campmlc, Deb now works in our Species File group at University of Illinois, which is amazing, and I copy her so she knows what is going on.

**Comment by campmlc** [comment]
Yes, the files are in a shared google folder at the moment. However, I am going through the process of loading names only to Arctos and using our validation tools to check, and I've caught a small number of errors that would be good to correct prior to load to GN.  I've spoken with Jenn Zaspel of TPT, and they are in process of hiring a postdoc to assist with taxonomy, so November would probably be fine as we can bring everyone on board. I can continue to load names to Arctos. We are creating a separate TPT (via Arctos) taxonomy source, and ideally we'd like to have the ability to auto-update this source from TPT via Global Names, rather than managing all the TPT taxonomy locally within Arctos.

**Comment by dimus** [comment]
Sounds perfect @campmlc! Then we will import it in November. It would be the best to have a stable "permanent" link to the latest versions of the data when it is ready. That would make regular updating of Parasite Tracker for GN much easier.

**Comment by campmlc** [comment]
Checking back in- can we arrange a call to discuss?

**Comment by dimus** [comment]
Yes @campmlc, Tuesdays and Thursdays are best for me, and about any time that is reasonable for CST is good.

**Comment by campmlc** [comment]
Can you join us today at 10:30 CST?

**Comment by campmlc** [comment]
If so, please email for joining instructions. If not, we can set up a later time.
campmlc at unm.edu

**Comment by vijaybarve** [comment]
@dimus and @debpaul Can we meet on 19th Nov 2020 11AM ET?

**Comment by debpaul** [comment]
Works for me @vijaybarve. Please send a calendar link once it's confirmed. Thanks!

**Comment by dimus** [comment]
@campmlc, @vijaybarve finally, I am ready to ingest TCN taxonomy. Can you tell me if it is available now?

**Comment by Jegelewicz** [comment]
@dimus we are trying to figure out where a "stable 'permanent' link" should be created, as well as the best method for access to the experts who might update it. We have a TPT Taxonomy Team meeting on Mondays at 2pm MST. Any chance you come next week?

**Comment by dimus** [comment]
@Jegelewicz sure, I can join on Feb 22nd, Monday 2pm MST, 3pm CST

**Comment by debpaul** [comment]
> (to @dimus): We have a TPT Taxonomy Team meeting on Mondays at 2pm MST. Any chance you come next week?
@Jegelewicz will you send a link? I'd like to join please?

**Comment by Jegelewicz** [comment]
I'll send via email.

**Comment by dimus** [comment]
@Jegelewicz did you send the email? I did not seem to get one

**Comment by dimus** [comment]
looks like fields in csv files at https://github.com/njdowdy/tpt-taxonomy are not normalized. As all the data go into the same dataset, it would be great if you can aggregate data into one csv with the same fields names.

**Comment by Jegelewicz** [comment]
The Siphonaptera and Acari are in Darwin Core format - I suggest the others be put in that format as well. Using a published exchange standard will make it easier for anyone to use these resources.

**Comment by dimus** [comment]
@Jegelewicz, yes, I agree, Darwin Core terms would be a good choice

**Comment by dimus** [comment]
Trying to import fro standardized versions of csv, and I got the following problem in the process -- I try to use scientificName field to ingest a name-string but semantics of this field differ from set to set:

```text
✦ ❯ xsv select scientificName acari-standardized.csv |head
scientificName
Allothyridae
"Dicrogonatus gardineri (Warburton, 1912)"
"Dicrogonatus niger (Thon, 1906)"
"Haplothyrus expolitissimus (Berlese, 1923)"
"Holothyrus expolitissimus Berlese, 1923"
"Holothyrus australasiae Womersley, 1935"
"Holothyrus coccinella Gervais, 1842"
"Holothyrus constrictus Domrow, 1955"
"Lindothyrus rubellus Lehtinen, 1995"

dwca_hunter/tpt/tpt-taxonomy-main
✦ ❯ xsv select scientificName aves-standardized.csv |head
scientificName
Abeillia abeillei
Abeillia abeillei abeillei
Abeillia abeillei aurea
Abroscopus albogularis
Abroscopus albogularis albogularis
Abroscopus albogularis fulvifacies
Abroscopus albogularis hugonis
Abroscopus schisticeps
Abroscopus schisticeps flavimentalis

dwca_hunter/tpt/tpt-taxonomy-main
✦ ❯ xsv select scientificName ixodes-standardized.csv |head
scientificName
""
""
""
""
""
""
""
""
""
```

Looks like sometimes it contains 'full' names, sometimes 'canonical form', and, sometimes is not given.

It would help tremendously, if this field in all data-sets was given, and was, according too DwC: "The full scientific name, with authorship and date information if known". Is it possible to achieve?

**Comment by dimus** [comment]
Thanks to @EMTuckerLabUMMZ fixing scientificName field we do have a progress!

I was able to create a DwCA file this time
- scientificName field works well now
- I am able to create classification for names
- I was able to import names to https://resolver.globalnames.org, TPT is http://resolver.globalnames.org/data_sources/200 now!
- I will import TPT to https://verifier.globalnames.org as well in a few days.

However there are still some problems that need to be addressed for better results:

1. UTF-8 Encoding

File syphonaptera-standardized.csv and ixodes-standardized.csv appear to be in latin-1 (ISO-8859-1) encoding. I am fixing them
on the fly, but it would definitely be better to have all the files in UTF-8 encoding.

Something happened with acaria.csv during the conversation to scaria-standardized.csv and names that contain non-ASCII UTF-8
characters got "unknown UTF-8 character" � instead. For example,

acari.csv line 47766
GBIF,5173778,,,2180393,6924129,,,,"Neumania spinipes (O.F.Müller, 1776)",,,,,Checklist of Norwegian ticks and mites (Acari). Fauna norv. Ser. B 26:31-45,,,Animalia,Arthropoda,,,,Trombidiformes,,,,,Unionicolidae,,,,Neumania,,spinipes,,species,,"(O.F.Müller, 1776)",,,accepted,,,Neumania spinipes

acari-standardised.csv
GBIF,5173778,,,2180393,6924129,,,,"Neumania spinipes (O.F.M�_ller, 1776)",,,,,Checklist of Norwegian ticks and mites (Acari). Fauna norv. Ser. B 26:31-45,,,Animalia,Arthropoda,,,,Trombidiformes,,,,,,Unionicolidae,,,,Neumania,,spinipes,,species,,"(O.F.M�_ller, 1776)",,,accepted,,,Neumania spinipes

It seems this problem appears only with acari-standardized, but it repeats in 1483 lines

2. taxonID field
 
Sometimes taxonID is empty, and if it exists, it is not a unique number. I generated gn_### id for now, but I am not able to use acceptedNameUsageID, because during compilation of data into data-source IDs, as they are given now, will create a confusion. So for now synonyms are not shown as synonyms.

I wonder if it is possible to convert IDs to IDs with name-spaces for each checklists, so that the IDs work correctly as references like in
acceptedNameUsageID, and are unique across all checklists. For example "acari" names would have IDs like "acari_12345" and so on.

Do you think it is possible to do? Hopefully we can find a path to provide synonymy.

**Comment by dimus** [comment]
Another thing I just noticed is that authorship in ixodes files often does not follow convensions. For example

```
Rhipicentor nuttalli COOPER ROBINSON
```

Capitalization is not a big problem, but luck of punctuation makes it hard to parse the name correctly. Any of the following would suffice:

```
Rhipicentor nuttalli COOPER, ROBINSON 
Rhipicentor nuttalli COOPER & ROBINSON 
Rhipicentor nuttalli COOPER and ROBINSON 
```

When authors are parsed it is expected to have different authors to be separated by commas, and optionally the last author can be separated by `and` or `&`:

```
Aus bus A. Bonn, B. Clack, Jeden & Moore
```

**Comment by EMTuckerLabUMMZ** [comment]
Regarding the author formatting - These variations are due to receiving the
names in different formats, exported from different databases. It would be
nice just from a readability and sorting perspective to all be
formatted the same way, but I think that may require a smart script of some
kind. It would take quite a lot of time to standardize all the authorities
by hand...

----------
Erika Tucker, PhD (s*he/her/hers*)
Milwaukee Public Museum | Digitization Taxonomy Manager
<https://parasitetracker.org/>
Biodiversity Outreach Network (BON) <https://www.bon-earth.org/> | Research
Associate <https://www.bon-earth.org/about-us>
*Entomologist, Museum Specialist, & Researcher*
Check it out!* Bug.News <http://Bug.News>*
ORCiD 0000-0002-8822-2315 <https://orcid.org/0000-0002-8822-2315>

**Comment by dimus** [comment]
@EMTuckerLabUMMZ lets leave authors as they are for now then, hopefully ixodes data can be fixed with time.

**Comment by debpaul** [comment]
@EMTuckerLabUMMZ 
> Regarding the author formatting - These variations are due to receiving the names in different formats, exported from different databases. It would be nice just from a readability and sorting perspective to all be formatted the same way, but I think that may require a smart script of some kind. It would take quite a lot of time to standardize all the authorities by hand...

Would this be the kind of task that "community curators" could manage? If we had a group who manages ixodes names, they could standardize these? Asking to understand where some community curation might be possible, might help. (Specifically for ixodes and in general).

**Comment by dimus** [comment]
I looked at the authorship in the ixodes file. There are ~1000 authorships, and there is no one clear rule that can describe them all,. However most of them can be fixed by just adding comma after a word. How about if I make a script that adds commas and fixes capitalization where possible, and someone would go through the result manually and fixes places where the script could not make the right choice? I would estimate it into 2-3 hours of manual labor

**Comment by EMTuckerLabUMMZ** [comment]
A script sounds like a great idea. My hesitation with manually correcting
things is this file may only have 1,000 author entries, which would be
doable, but other files may have 40,000 or even 400,000 authors depending
on the taxa... Just my opinion :)

----------
Erika Tucker, PhD (s*he/her/hers*)
Milwaukee Public Museum | Digitization Taxonomy Manager
<https://parasitetracker.org/>
Biodiversity Outreach Network (BON) <https://www.bon-earth.org/> | Research
Associate <https://www.bon-earth.org/about-us>
*Entomologist, Museum Specialist, & Researcher*
Check it out!* Bug.News <http://Bug.News>*
ORCiD 0000-0002-8822-2315 <https://orcid.org/0000-0002-8822-2315>

**Comment by dimus** [comment]
@EMTuckerLabUMMZ how about this: I will add a new field, for example AuthorshipNormalized, or something like that, to the standardised version of the ixodes, and add results from the script there. I can either just send it to you by email, or create a pull request.

**Comment by EMTuckerLabUMMZ** [comment]
Sounds good! Thanks Dmitry!

----------
Erika Tucker, PhD (s*he/her/hers*)
Milwaukee Public Museum | Digitization Taxonomy Manager
<https://parasitetracker.org/>
Biodiversity Outreach Network (BON) <https://www.bon-earth.org/> | Research
Associate <https://www.bon-earth.org/about-us>
*Entomologist, Museum Specialist, & Researcher*
Check it out!* Bug.News <http://Bug.News>*
ORCiD 0000-0002-8822-2315 <https://orcid.org/0000-0002-8822-2315>
